### PR TITLE
Make usable as a ZSH plugin (simply adds to PATH) This also hard codes the python scripts with the abspath to the osx python interpreter, that way homebrew doesn't get in the way and such.

### PR DESCRIPTION
**Pull request by akatrevorjay**
One rather large change is the removal of extensions, since they were a bit counter-intuitive, at least for me.

### CODE REVIEW

**Comment by liyanage** [comment]
Thanks for the patch. Unfortunately it is a bit too disruptive so I won't merge most of it for these reasons:
- I prefer the extensions present, also if anybody cloned or copied them and uses them in scripts, it will cause a lot of breakage for no real benefit
- The indirection for the Python interpreter is actually a desirable feature, precisely so people can override it with things like homebrew

I'll be happy to take the "macosx-shell-scripts.plugin.zsh" patch, can you explain a bit more what that does?

**Comment by akatrevorjay** [comment]
I had a feeling that may be the case. To be honest I only created the PR in quick hindsight!

Sure, the plugin.zsh file just allows zsh frameworks to download and include these scripts into their path easily, some examples:

https://github.com/zplug/zplug
https://github.com/tarjoilija/zgen

**Comment by akatrevorjay** [comment]
If you want you can cherry pick it out, or I can split it apart sometime over the next day or two.

**Comment by akatrevorjay** [comment]
IMO, you may want to consider also nabbing the force to use OSX python portions (shebang hard-code), as any other python installed will not work with most of these. If you install python from brew for instance, or even pyenv, they won't work as they rely on compiled python extensions that come with osx (but I'm sure you know all of this, I'm more posting the reasoning for posterity)

File: macosx-shell-scripts.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/zsh
+path+=("${0:h}")
```
Comment (akatrevorjay) [inline macosx-shell-scripts.plugin.zsh:2]: This file is loaded automagically via any ZSH framework upon instantiation. ZSH treats path as an array view on your standard PATH variable, so this just appends PATH with a single element. $0 is of course the script name, and the modifier `:h` is equivalent to `dirname`

**Pull request closed without merging**